cinema/serializers.py

```diff
@@ -1 +1,101 @@
-# write serializers here
+from rest_framework import serializers
+
+from cinema.models import CinemaHall, Genre, Actor, Movie, MovieSession
+
+
+class ActorSerializer(serializers.ModelSerializer):
+    class Meta:
+        model = Actor
+        fields = [
+            "id",
+            "first_name",
+            "last_name",
+            "full_name"
+        ]
+
+
+class GenreSerializer(serializers.ModelSerializer):
+    class Meta:
+        model = Genre
+        fields = ["id", "name"]
+
+
+class CinemaHallSerializer(serializers.ModelSerializer):
+    class Meta:
+        model = CinemaHall
+        fields = ["id", "name", "rows", "seats_in_row", "capacity"]
+
+
+class MovieSerializer(serializers.ModelSerializer):
+    class Meta:
+        model = Movie
+        fields = [
+            "title",
+            "description",
+            "duration",
+            "genres",
+            "actors"
+        ]
+
+
+class MovieListSerializer(MovieSerializer):
+    genres = serializers.SlugRelatedField(
+        many=True, read_only=True, slug_field="name"
+    )
+    actors = serializers.SlugRelatedField(
+        many=True, read_only=True, slug_field="full_name"
+    )
+
+
+class MovieDetailSerializer(MovieSerializer):
+    genres = GenreSerializer(many=True)
+    actors = ActorSerializer(many=True)
+
+
+class MovieSessionSerializer(serializers.ModelSerializer):
+    class Meta:
+        model = MovieSession
+        fields = [
+            "id",
+            "show_time",
+            "movie",
+            "cinema_hall"
+        ]
+
+
+class MovieSessionListSerializer(MovieSessionSerializer):
+    movie_title = serializers.CharField(source="movie.title", max_length=255)
+    cinema_hall_name = serializers.CharField(
+        source="cinema_hall.name", max_length=255
+    )
+    cinema_hall_capacity = serializers.IntegerField(
+        source="cinema_hall.capacity"
+    )
+
+    class Meta(MovieSessionSerializer.Meta):
+        fields = (
+            "id",
+            "show_time",
+            "movie_title",
+            "cinema_hall_name",
+            "cinema_hall_capacity",
+        )
+        read_only_fields = [
+            "movie_title",
+            "cinema_hall_name",
+            "cinema_hall_capacity",
+        ]
+
+
+class MovieSessionDetailSerializer(serializers.ModelSerializer):
```

Comment:
why don't u inherit from MovieSessionSerializer to avoid implementing meta class here?

Reply:
> why don't u inherit from MovieSessionSerializer to avoid implementing meta class here?

Agree. It is redundant